comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +50,86 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+            long ret = 0;
+            foreach (string file in Traverse(directory, true, false))
+            {//add the size of every file to the return variable
+                ret += GetFileSize(file);
+            }
+            return ret;
 		}
 
 		// Return the number of files (not counting directories) below the given directory
 		public static int CountFiles(string directory)
-		{
-			throw new NotImplementedException();
+		{//return the amount of entries when searching for only files
+            return Traverse(directory, true, false).Count;
```

Comment:
Neat! The `System.Linq` namespace has some useful methods which you could use to make some of the other functions similarly concise